informers.go

```diff
@@ -23,14 +43,122 @@ func getVirtCli() (kubecli.KubevirtClient, error) {
 	return virtCli, err
 }
 
-func getVmiInformer() (cache.SharedIndexInformer, error) {
-	virtCli, err := getVirtCli()
+func getVmiInformer(virtCli kubecli.KubevirtClient) (cache.SharedIndexInformer, error) {
+	listWatcher := cache.NewListWatchFromClient(virtCli.RestClient(), "virtualmachineinstances", namespace, fields.Everything())
+	vmiInformer := cache.NewSharedIndexInformer(listWatcher, &k6tv1.VirtualMachineInstance{}, 1*time.Hour, cache.Indexers{})
+	
+	vmiInformer.AddEventHandler(cache.ResourceEventHandlerFuncs {
+		UpdateFunc: handleVmiUpdate,
+		DeleteFunc: handleVmiDeletion,
+	})
+
+	return vmiInformer, nil
+}
+
+func updateMachineTypes(virtCli kubecli.KubevirtClient) error {
+	vmList, err := virtCli.VirtualMachine(namespace).List(&k8sv1.ListOptions{})
 	if err != nil {
-		return nil, err
+		return err
+	}
+	for _, vm := range vmList.Items {
+		machineType := vm.Spec.Template.Spec.Domain.Machine.Type
+		machineTypeSubstrings := strings.Split(machineType, "-")
+		version := machineTypeSubstrings[2]
+		if len(machineTypeSubstrings) != 3 {
+			return nil
+		}
+		
+		if strings.Contains(version, "rhel") && version < "rhel9.0.0" {
+			machineTypeSubstrings[2] = latestMachineTypeVersion
+			machineType = strings.Join(machineTypeSubstrings, "-")
+			updateMachineType := fmt.Sprintf(`{"spec": {"template": {"spec": {"domain": {"machine": {"type": "%s"}}}}}}`, machineType)
+			
+			_, err = virtCli.VirtualMachine(vm.Namespace).Patch(vm.Name, types.StrategicMergePatchType, []byte(updateMachineType), &k8sv1.PatchOptions{})
+			if err != nil {
+				return err
+			}
+			
+			// add label to running VMs that a restart is required for change to take place
+			if vm.Status.Ready {
```

Comment:
```suggestion
			if vm.Status.Created {
```

There's a slight difference between `Created`, which means that a VMI object exists for this VM, and `Ready`, which means that the VMI is up and ready (e.g. booted, connected, etc).

If we use `Ready` here we might miss a VM that had just started but did not fully boot yet. However, at this point, the machine type is already determined.